src/components/sbb-datepicker/sbb-datepicker.tsx

```diff
@@ -0,0 +1,213 @@
+  public datepickerPropChanged(newValue: any, oldValue: any): void {
+    if (newValue !== oldValue) {
+      this.datePickerUpdated.emit();
+    }
+  }
+
+  @Watch('_inputElement')
+  public registerInputElement(newValue: HTMLInputElement, oldValue: HTMLInputElement): void {
+    if (newValue !== oldValue) {
+      this._datePickerController?.abort();
+      this._datePickerController = new AbortController();
+
+      if (!this._inputElement) {
+        return;
+      }
+
+      this._inputObserver?.disconnect();
+      this._inputObserver.observe(this._inputElement, {
+        attributeFilter: ['disabled', 'readonly', 'min', 'max'],
+      });
+
+      this._inputElement.type = 'text';
+      this._inputElement.placeholder = i18nDatePickerPlaceholder[this._currentLanguage];
+
+      this._inputElement.addEventListener(
+        'input',
+        (event: Event) => this._preventCharInsert(event),
+        { signal: this._datePickerController.signal }
+      );
+      this._inputElement.addEventListener('change', (event: Event) => this._valueChanged(event), {
+        signal: this._datePickerController.signal,
+      });
+    }
+  }
+
+  @Listen('sbbLanguageChange', { target: 'document' })
+  public handleLanguageChange(event: SbbLanguageChangeEvent): void {
+    this._currentLanguage = event.detail;
+  }
+
+  /** Gets the input value with the correct date format. */
+  @Method() public async getValueAsDate(): Promise<Date> {
+    return this._dateAdapter.formatValueAsDate(this._formatValue(this._inputElement?.value));
+  }
+
+  /** Set the input value to the correctly formatted value. */
+  @Method() public async setValueAsDate(date: Date): Promise<void> {
+    const newValue: string = this._formatValue(
+      `${date.getDate()}.${date.getMonth() + 1}.${date.getFullYear()}`
+    );
+    await this._formatAndUpdateValue(newValue);
+  }
+
+  private _datePickerController: AbortController;
+
+  private _inputObserver = new MutationObserver(this._onInputPropertiesChange.bind(this));
+
+  private _dateAdapter = new NativeDateAdapter();
+
+  public connectedCallback(): void {
+    this._inputElement = getInput(this._element, this.input);
+  }
+
+  public componentDidLoad(): void {
+    this.inputUpdated.emit({
+      disabled: this._inputElement?.disabled,
+      readonly: this._inputElement?.readOnly,
+      min: this._inputElement?.min,
+      max: this._inputElement?.max,
+    });
+  }
+
+  public disconnectedCallback(): void {
+    this._inputObserver?.disconnect();
+    this._datePickerController?.abort();
+  }
+
+  private _formatValue(value: string): string {
+    const match: RegExpMatchArray = value?.match(REGEX);
+    if (!match || match.length <= 2) {
+      return value;
+    }
+
+    const day: string = match[1].padStart(2, '0');
+    const month: string = match[2].padStart(2, '0');
+    let year: number = +match[3];
+    if (year < 100 && year >= 0) {
+      year += 1900;
+    }
+    return `${day}.${month}.${year}`;
```

Comment:
should we not use DateTimeFormat with options { year: 'numeric', month: '2-digit', day: '2-digit' }? maybe not, I don't really know :D

Reply:
Currently we're formatting the date as dd.mm.yyyy, and DateTimeFormat doesn't really let you choose the separator. In this way with lang="en" we would have mm/dd/yyyy